interface/main/holidays/Holidays_Storage.php

```diff
@@ -0,0 +1,137 @@
+        $res=sqlStatement($sql, array(self::CALENDAR_CATEGORY_HOLIDAY,self::CALENDAR_CATEGORY_CLOSED,$start_date,$end_date) );
+        while ($row = sqlFetchArray($res)) {
+            $holidays[] = $row['pc_eventDate'];
+        }
+        return $holidays;
+
+    }
+
+    /**
+     * From an array of holidays creates a row that will be inserted as an event to be used in the calendar
+     * The holidays array must contai the date=>DD/MM/YYY, description=>"string"
+     * @param array $holidays
+     */
+    public function create_events(array $holidays){
+        $deleted = false;
+        foreach ($holidays as $holiday){
+            if(!$deleted){
+                $this->delete_holiday_events();
+                $deleted = true;
+            }
+            $row=array(
+                self::CALENDAR_CATEGORY_HOLIDAY,//catgory
+                0,//authid
+                0,//pid
+                $holiday['description'],//title
+                $holiday['date'],//date
+                86400,//duration all day in seconds
+                "a:6:{s:17:\"event_repeat_freq\";s:1:\"0\";s:22:\"event_repeat_freq_type\";s:1:\"0\";s:19:\"event_repeat_on_num\";s:1:\"1\";s:19:\"event_repeat_on_day\";s:1:\"0\";s:20:\"event_repeat_on_freq\";s:1:\"0\";s:6:\"exdate\";s:0:\"\";}",
+                1,//allday
+                1,//status
+                $_SESSION['pc_facility'],//facility
+                2 //SHARING_PUBLIC
+            );
+
+            $pc_eid = sqlInsert("INSERT INTO openemr_postcalendar_events ( " .
+                "pc_catid,  pc_aid, pc_pid, pc_title, pc_time, " .
+                "pc_eventDate,  pc_duration, " . "pc_recurrspec,  pc_alldayevent, " . " pc_eventstatus, pc_facility,pc_sharing" .
+                ") VALUES (?,?,?,?,NOW(),?,?,?,?,?,?,?)",
+                $row
+            );
+        }
+        return true;
+
+    }
+
+    /**
+     * This function opend the $file(csv) and parses it to insert the values in the calendar_external so later they can be imported as events
+     * csv format -> date,description
+     * Example:
+     * 2016/12/24,Christmas
+     * @param $file (string containing the file name)
+     */
+    public function import_holidays($file){
+        $data = array();
+        $handle = fopen($file,"r");
+        $deleted = false;
+        do {
+            if ($data[0]) {
+                if(!$deleted){
+                    $this->delete_calendar_external();
+                    $deleted = true;
+                }
+
+                $row=array($data[0],$data[1]);
+                sqlInsert("INSERT INTO ".escape_table_name(self::TABLE_NAME) . "(id,date,description,source)"." VALUES ('',?,?,'csv')",$row);
+
+            }
+        } while ($data = fgetcsv($handle,1000,",","'"));
+        return true;
+    }
+
+    private function delete_calendar_external(){
+        $sql = "DELETE FROM ".escape_table_name(self::TABLE_NAME);
+        $res=sqlStatement($sql);
```

Comment:
Would it make sense to instead use Truncate so the auto counting index for id is reset?